ext/winevt/winevt_utils.cpp

```diff
@@ -500,3 +500,184 @@ get_description(EVT_HANDLE handle)
 
   return _wcsdup(result.data());
 }
+
+VALUE
+render_system_event(EVT_HANDLE hEvent)
+{
+  DWORD status = ERROR_SUCCESS;
+  EVT_HANDLE hContext = NULL;
+  DWORD dwBufferSize = 0;
+  DWORD dwBufferUsed = 0;
+  DWORD dwPropertyCount = 0;
+  VALUE vRenderedValues;
+  PEVT_VARIANT pRenderedValues = NULL;
+  WCHAR wsGuid[50];
+  LPSTR pwsSid = NULL;
+  ULONGLONG ullTimeStamp = 0;
+  ULONGLONG ullNanoseconds = 0;
+  SYSTEMTIME st;
+  FILETIME ft;
+  CHAR buffer[32];
+  VALUE rbstr;
+  DWORD EventID;
+  VALUE hash = rb_hash_new();
+
+  hContext = EvtCreateRenderContext(0, NULL, EvtRenderContextSystem);
+  if (NULL == hContext) {
+    rb_raise(rb_eWinevtQueryError,
+             "Failed to create renderContext with %lu\n", GetLastError());
+  }
+
+  if (!EvtRender(hContext,
+                 hEvent,
+                 EvtRenderEventValues,
+                 dwBufferSize,
+                 pRenderedValues,
+                 &dwBufferUsed,
+                 &dwPropertyCount)) {
+    if (ERROR_INSUFFICIENT_BUFFER == (status = GetLastError())) {
```

Comment:
Please don't include assignment in `if` condition.

Reply:
Fixed in https://github.com/fluent-plugins-nursery/winevt_c/pull/11/commits/32ac13a5f0a786b9aa064f25f5695ff5aeee52ba.